src/producer.rs

```diff
@@ -347,37 +351,65 @@ fn _register_function(function_name: String, extension_info: &mut (Vec<extension
 
 }
 
+/// Return Substrait scalar function with two arguments
+pub fn make_binary_op_scalar_func(lhs: &Expression, rhs: &Expression, op: Operator, extension_info: &mut (Vec<extensions::SimpleExtensionDeclaration>, HashMap<String, u32>)) -> Expression {
+    let function_name = operator_to_name(op).to_string().to_lowercase();
+    let function_anchor = _register_function(function_name, extension_info);
+    Expression {
+        rex_type: Some(RexType::ScalarFunction(ScalarFunction {
+            function_reference: function_anchor,
+            arguments: vec![
+                FunctionArgument {
+                    arg_type: Some(ArgType::Value(lhs.clone())),
+                },
+                FunctionArgument {
+                    arg_type: Some(ArgType::Value(rhs.clone())),
+                },
+            ],
+            output_type: None,
+            args: vec![],
+        })),
+    }
+}
+
 /// Convert DataFusion Expr to Substrait Rex
 pub fn to_substrait_rex(expr: &Expr, schema: &DFSchemaRef, extension_info: &mut (Vec<extensions::SimpleExtensionDeclaration>, HashMap<String, u32>)) -> Result<Expression> {
     match expr {
+        Expr::Between { expr, negated, low, high } => {
+            match negated {
+                true => {
```

Comment:
nit: probably more idiomatic to use an `if` statement, rather than match on a boolean